### PR TITLE
Sidekiq scheduler

### PR DESCRIPTION
**Pull request by Joeper214**
for: https://skyarch.backlog.jp/view/NC-652

### CODE REVIEW

**Comment by Joeper214** [comment]
Please Review and Merge PR. @root-beer thanks! :bow:

**Comment by root-beer** [comment]
I think we should add the new migration file to do `drop_table :crono_jobs` instead of removing the migration file. What do you think? @Joeper214

**Comment by Joeper214** [comment]
Yes. That's a better idea. thanks.

**Comment by root-beer** [comment]
It seems not working...

**Comment by Joeper214** [comment]
I'll investigate it.

**Comment by Joeper214** [comment]
I think this [issue](https://github.com/skyarch-networks/skyhopper/pull/319#issuecomment-243070016) seems to be addressed.

**Comment by Joeper214** [comment]
Please let me know if you have some comments on this PR. @root-beer  🙇

**Comment by Joeper214** [comment]
Okay. Maybe this error is too difficult to replicate and may cause certain problems in the future. I'll try to check for another solution to this.　すみません。:bow:

**Comment by root-beer** [comment]
It's Okay. 👌

**Comment by root-beer** [comment]
I'll review it now.

File: app/jobs/server_runner.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#
```
Comment (Joeper214) [inline app/jobs/server_runner.rb]: I think We don't need this file as it is not been used. becuase the ServerStateWorker job is on [L117](https://github.com/skyarch-networks/skyhopper/pull/319/files#diff-f62f584c2dc1fb4f92f5d992b7b0bdaaR117)

Comment (Joeper214) [inline app/jobs/server_runner.rb]: すみません。

Comment (root-beer) [inline app/jobs/server_runner.rb]: No worries.

File: app/jobs/operation_worker.rb [hunk 2]
```diff
@@ -39,86 +48,74 @@ def perform
...
+      log_msg = "Stopped: #{instance.physical_id}. As Scheduled on #{params[:repeats]}"
 
-      log = InfrastructureLog.create(infrastructure_id: resource.infrastructure_id, user_id: user_id, details: log_msg, status: true)
+      log = InfrastructureLog.create(infrastructure_id: params[:resource].infrastructure_id, user_id: params[:user], details: log_msg, status: true)
       ws.push_as_json({message: log.details, status: log.status, timestamp: Time.zone.now.to_s})
     end
   end
 
+  Sidekiq::Cron::Job.create(name: 'OperationWorker worker - every 3min', cron: '*/3 * * * *', class: 'OperationWorker')
+  Sidekiq::Cron::Job.create(name: 'ServerRunner for ServerStateWorker - every 3min', cron: '*/3 * * * *', class: 'ServerStateWorker')
```
Comment (root-beer) [inline app/jobs/operation_worker.rb]: Why don't you separate this part in another file like a .yml?

e.g.) https://github.com/ondrejbartas/sidekiq-cron#adding-cron-job

**Comment by Joeper214** [comment]
I tried this methods before but it wasn't working. I'll retry now. Thanks.

File: config/initializers/sidekiq.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+schedule_file = "config/schedule.yml"
+
+if File.exists?(schedule_file) && Sidekiq.server?
```
Comment (skyarch-networks) [inline config/initializers/sidekiq.rb]: __[rubocop]__
* Lint/DeprecatedClassMethods: `File.exists?` is deprecated in favor of `File.exist?`.

**Comment by skyarch-networks** [comment]
Click here if you want to create rubocop's auto-correct Pull Request 
[Auto Correct by SideCI](https://sideci.com/gh/31754199/issues/pull_request/319/diff/ccc369a532fcc10a803ced962f49ed866685f4c3?open_modal=1#rubocop)

See the documentation if you want to get more information about auto-correct.
[Document](https://sideci.com/ja/docs/autofix-rubocop)

**Comment by Joeper214** [comment]
I Think it's working now. Let me know. Thanks! @root-beer 🙇

**Comment by root-beer** [comment]
It works! 😄

**Comment by root-beer** [comment]
LGTM! Thanks. @Joeper214 :bow:

**Comment by Joeper214** [comment]
:bow:

**Pull request merged**